File: Bogosort/Go/jcla1/bogosort.go
package bogosort

import "math/rand"

func Bogosort(arr []int) []int {
	for !isSorted(arr) {
		fisherYates(arr)
	}

	return arr
}

func isSorted(arr []int) bool {
	for i := 1; i < len(arr); i++ {
		if arr[i-1] > arr[i] {
			return false
		}
	}

	return true
}

// This function is just for shuffling,
// not part of the actual algorithm
func fisherYates(arr []int) []int {
	m := len(arr)
	var i int

	for m != 0 {
		i = rand.Intn(m)
		m -= 1
		arr[i], arr[m] = arr[m], arr[i]
	}

	return arr
}

File: Bogosort/Java/jcla1/Bogosort.java
import java.util.Random;

public class Bogosort {
    public static int[] sort(int[] arr) {
        while (!isSorted(arr)) {
            fisher_yates(arr);
        }
        return arr;
    }

    private static boolean isSorted(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < arr[i-1]) return false;
        }
        return true;
    }

    // Soley used for shuffling, not part of actual algorithm
    private static void fisher_yates(int[] arr) {
        Random rand = new Random();
        int m = arr.length;
        int tmp, i;

        while (m != 0) {
            i = rand.nextInt(m--);
            tmp = arr[m];
            arr[m] = arr[i];
            arr[i] = tmp;
        }
    }
}